From 00192c17937fabc7b15191325b66b0616e2a4b77 Mon Sep 17 00:00:00 2001
From: Vishnu Banavath <vishnu.banavath@arm.com>
Date: Fri, 17 Dec 2021 19:49:02 +0000
Subject: [PATCH 22/27] efi_loader: populate ESRT table if EFI_ESRT config
 option is set

This change is to call efi_esrt_populate function if CONFIG_EFI_ESRT
is set. This will populte esrt table with firmware image info

Signed-off-by: Vishnu Banavath <vishnu.banavath@arm.com>
Signed-off-by: Rui Miguel Silva <rui.silva@linaro.org>
---
 lib/efi_loader/efi_capsule.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/lib/efi_loader/efi_capsule.c b/lib/efi_loader/efi_capsule.c
index 1475bef7030b..9ffc46375341 100644
--- a/lib/efi_loader/efi_capsule.c
+++ b/lib/efi_loader/efi_capsule.c
@@ -675,6 +675,13 @@ efi_status_t __efi_runtime EFIAPI efi_update_capsule(
 			ret = EFI_SUCCESS;
 		}
 
+		if (IS_ENABLED(CONFIG_EFI_ESRT)) {
+			/* Rebuild the ESRT to reflect any updated FW images. */
+			ret = efi_esrt_populate();
+	               if (ret != EFI_SUCCESS)
+				log_warning("EFI Capsule: failed to update ESRT\n");
+	       }
+
 		goto out;
 #endif
 
-- 
2.35.1

